doc/arkode/guide/source/Usage/MRIStep/User_callable.rst

```diff
@@ -249,20 +290,20 @@ Nonlinear solver interface functions
    This function specifies the ``SUNNonlinearSolver`` object
    that MRIStep should use for implicit stage solves.
 
-   **Arguments:**
-      * *arkode_mem* -- pointer to the MRIStep memory block.
-      * *NLS* -- the ``SUNNonlinearSolver`` object to use.
+   :param arkode_mem: pointer to the MRIStep memory block.
+   :param NLS: the ``SUNNonlinearSolver`` object to use.
 
-   **Return value:**
-      * *ARK_SUCCESS*   if successful
-      * *ARK_MEM_NULL*  if the MRIStep memory was ``NULL``
-      * *ARK_MEM_FAIL*  if there was a memory allocation failure
-      * *ARK_ILL_INPUT* if MRIStep is incompatible with the
-        provided *NLS* input object.
+   :retval ARK_SUCCESS:   if successful
+   :retval ARK_MEM_NULL:  if the MRIStep memory was ``NULL``
+   :retval ARK_MEM_FAIL:  if there was a memory allocation failure
+   :retval ARK_ILL_INPUT: if MRIStep is incompatible with the
+                          provided *NLS* input object.
+
+   .. note::
 
-   **Notes:**  MRIStep will use the Newton ``SUNNonlinearSolver`` module by
-   default; a call to this routine replaces that module with the
-   supplied *NLS* object.
+      MRIStep will use the Newton ``SUNNonlinearSolver`` module by
+      default; a call to this routine replaces that module with the
+      supplied *NLS* object.
```

Comment:
Is this true? I realize this is the same text as in ARKstep, and with ARKstep, I recall not setting a NLS resulted in a error (a seg fault!). Perhaps this has changed since I last tried.

Reply:
This is accurate -- both here and in ARKStep.  Also, aside from me updating the formatting of this file, nothing about this function was modified in this PR.  I don't know what was going on in the instance that you reference, but most of our implicit ARKStep and MRIStep examples are do not bother with manually attaching a SUNNonlinearSolver.  

In both `MRIStepCreate` and `MRIStepReInit` we have always allocated the `SUNNonlinSol_Newton` by default, and then freed/replaced this if a user called `MRIStepSetNonlinearSolver`.

Reply:
That's right. I was thinking of the linear solver. There's no default, and if it's not set by a user, that results in a seg fault.